ast.go

```diff
@@ -429,4 +462,18 @@ func (g *graph) resolveFQN(e Entity) string {
 	return e.FullyQualifiedName()
 }
 
+func assignDependent(ft FieldType, parent Entity) {
+	if ft.IsEnum() {
+		enum := ft.Enum()
+		if enum.File().Name() != parent.Name() {
```

Comment:
Can you explain the reason for this check?

Reply:
initially i was thinking if the enum/message comes from the same file as the parent then the parent already has a reference to it but this isn't always true so i'll update the check.

Reply:
so for the updated logic, i just check if the enum/message's parent entity is the same as `parent` since if it is then the necessary dependents are already factored in and it can be skipped